src/main/java/com/iota/iri/IXI.java

```diff
@@ -70,9 +69,9 @@ public void init(String rootDir) throws Exception {
     private void registerRecursive(final Path root) throws IOException {
         Files.walkFileTree(root, EnumSet.allOf(FileVisitOption.class), MAX_TREE_DEPTH, new SimpleFileVisitor<Path>() {
             @Override
-            public FileVisitResult preVisitDirectory(Path modulePath, BasicFileAttributes attrs) throws IOException {
+            public FileVisitResult preVisitDirectory(Path modulePath, BasicFileAttributes attrs) {
                 watch(modulePath);
-                if (modulePath != rootPath) {
+                if (!modulePath.equals(rootPath)) {
```

Comment:
I prefer to always use `Objects.equals()` since it will never throw null pointer exception.
Only exception is if you want to have the exception thrown or that the first argument is a constant.

Reply:
I don't understand. That is what i am doing here. I replaced != by !modulePath.equals(rootPath)

Reply:
I meant doing `Objects.equals(modulePath,rootPath)`.
Looking at it again I think it is fine that a `nullPointerException`  may occur. So I will approve.